### ISSUE TITLE
[Bug]: Frontend::getSiteIdForDocument returns wrong id / wrong document public url

### ISSUE THREAD

**Issue by simonkey** [pimcore/pimcore#17871]
### Pimcore version

11.4.x

### Steps to reproduce

The problem 
Add two sites where one site key beginning is contained in the other one 
- Site key 1: abc
- Site key 2: abcdef

In this case the Pimcore\Tool\Frontend::getSiteIdForDocument(...) returns always the first site. Changing the order of the sites doesn't effect the result, because are sorted by length descending.

The problem is in [Pimcore\Tool\Frontend in line #68](https://github.com/pimcore/pimcore/blob/11.x/lib/Tool/Frontend.php#L68). The check returns a false positive, because it's not checked if the match is exact.

The issue was introduced with the commit https://github.com/pimcore/pimcore/commit/c424e036a0df053c1d0063da1db9d0e28fcb99ca (PR https://github.com/pimcore/pimcore/pull/14706)

### Actual Behavior

In this case the Pimcore\Tool\Frontend::getSiteIdForDocument(...) returns always the first site. Changing the order of the sites doesn't effect the result, because are sorted by length descending.

As a result the public url for the document in the admin area is wrong.

Current code:

```
        foreach ($siteMapping as $sitePath => $id) {
            if (str_starts_with($document->getRealFullPath(), $sitePath) || $do) {
                return $id;
            }
        }
```

### Expected Behavior

Pimcore\Tool\Frontend::getSiteIdForDocument(...) should return always the correct site and not the first matching one.

Improved code:

```
        $documentRealFullPath = $document->getRealFullPath();
        foreach ($siteMapping as $sitePath => $id) {
            if (str_starts_with($documentRealFullPath, $sitePath.'/') || $documentRealFullPath === $sitePath) {
                return $id;
            }
        }
```

**Comment by simonkey** [comment]
Any update, if this issue will be resolved in the future? There is already a pull request with a solution to the problem.

**Comment by robertSt7** [comment]
fixed by #17872
